lib/metasploit/framework/login_scanner/base.rb

```diff
@@ -57,8 +57,8 @@ module Base
                     inclusion: { in: [true, false] }
 
           validates :bruteforce_speed,
-                    presence: false,
                     numericality: {
+                      allow_nil: true,
```

Comment:
IMO this would be easier to read using conditional validation (e.g. unless: "bruteforce_speed.nil?" versus allow_nil: true). That way you are saying I want these attributes of numericality checked, unless the field is blank, versus slipping the allowance of the nil into the numeric checks.